README.md

```diff
@@ -1,25 +1,36 @@
-# Clean up
+# Clean-up
 
-This library helps to free up all desired resources or do all clean-up operations by a single call.
+This library helps to free up all desired resources and clean-up operations by a single call.
 
 By using this library, we will make sure that all operations, will be performed if it is possible.
 That means, if any exceptions occurs during clean-up operations of a resource, it will
-not prevent clean up of other resources.
+not prevent clean-up of other resources.
 
 ### Sample Usage
 
+`Cleanups` accepts a list of `AutoCloseable` and `RunnableWithException`.
+Upon calling `doAll()` all `AutoCloseable` will be called and all `RunnableWithException` statements will be executed.
+
 An example usage may look like this:
 
 ```java
-Cleanups.of(...)
-        .and(...)
-        .and(...)
-        .and(...)
+List<FlowTracker> appThreads = new List<>();
```

Comment:
Replace 
FlowTeacker > Thread

Reply:
After final change, Thread is not good because it only provides 'run' method which is different from 'close' method.